scout/server/blueprints/variants/views.py

```diff
@@ -62,15 +59,32 @@ def variants(institute_id, case_name):
             'gene_panels': clinical_filter_panels
              })
 
+    user_obj = store.user(current_user.email)
     if(request.method == "POST"):
+        # If special filter buttons were selected:
         if bool(request.form.get('clinical_filter')):
             form = FiltersForm(clinical_filter)
             form.csrf_token = request.args.get('csrf_token')
+        elif bool(request.form.get('save_filter')):
+            # The form should be applied and remain set the page after saving
+            form = FiltersForm(request.form)
+            # Stash the filter to db to make available for this institute
+            filter_obj = request.form
+            store.stash_filter(filter_obj, institute_obj, case_obj, user_obj, category='snv')
+        elif bool(request.form.get('load_filter')):
+            filter_display_name = request.form.get('filters')
+            filter_obj = store.retrieve_filter(filter_display_name)
```

Comment:
What if you have more than one filter with the same display name?

Reply:
Thanks, thats a bad old variable name. The functions actually use a unique id for retrieval. Fixed!